### DISCUSSION TITLE
Heads up, `fzf` is going to allow wrapping long lines

### DISCUSSION THREAD

**Discussion by tuurep** [papis/papis#884] [Show and tell]
https://github.com/junegunn/fzf/releases/tag/v0.54.0

I'll be interested to try this with the file picker, because pretty much always my document titles are longer than fit the screen, which is like *the* downside of using `fzf` over the default.

**Comment by tuurep** [comment]
As I'm working on some docs improvements, I notice that the docs are quite inaccurate regarding this:

> However, a disadvantage is that [`fzf`] is restricted to one-line entries. 

> `fzf` has the disadvantage that it does not support multiline output...

I'm interested in fixing this too, but will have to play around with what's actually possible/useful with the fzf multiline support + papis document entries.

What I ended up with was a keybind to toggle line wrap, but it's not so nice because it wraps mid-word with no configurable break-points.

Would be useful to cook up an example that resembles the default Python picker multiline entry, but in fzf.

**Reply by alexfikl** [reply]
Yeah, the `fzf` picker docs haven't been updated since it was first introduced. You're welcome to update them and add examples and whatever else feels useful!

In general, the `fzf` picker could definitely use more love, so if you have the will and the knowledge it would be great to add something like multiline support :grin:

**Reply by tuurep** [reply]
Hmm okay

I was thinking maybe it's already achievable by setting the right fzf flags and `\0`s in templates, but never actually tried this

**Reply by alexfikl** [reply]
Maybe it is? If yes, we'll probably want to make use of it in the default template soon enough (fzf 0.54 is still rather new?). Until then, it would be nice to add it as an example to the docs somewhere.

**Reply by tuurep** [reply]
Yeah, although multi-line support was added in [0.53](https://github.com/junegunn/fzf/releases/tag/0.53.0) (June 2024), then line-wrap naturally followed from that.

Yes maybe there's potential for a multi-line template to be default. I'll take a look at this in the future, but let me know if you decide to try it out in the meanwhile.

Edit:

I also really like the `gap` flag that was added in [0.56](https://github.com/junegunn/fzf/releases/tag/v0.56.0) (2 days ago), makes it easier to create distinct blocks from multi-line entries

**Comment by tuurep** [comment]
Oh yes, as I was only talking about *line-wrap* in the opening post, here's a very relevant writeup regarding *multi-line* items in `fzf`:

https://junegunn.github.io/fzf/tips/processing-multi-line-items/
